dpc/src/program/execution.rs

```diff
@@ -15,15 +15,16 @@
 // along with the snarkVM library. If not, see <https://www.gnu.org/licenses/>.
 
 use crate::Parameters;
-use snarkvm_algorithms::SNARK;
+use snarkvm_algorithms::{merkle_tree::MerklePath, SNARK};
 use snarkvm_fields::{ConstraintFieldError, ToConstraintField};
 
 /// Program verifying key and proof.
 #[derive(Derivative)]
 #[derivative(Clone(bound = "S: SNARK"))]
-pub struct Execution<S: SNARK> {
+pub struct Execution<C: Parameters, S: SNARK> {
```

Comment:
do you think it is possible AND reasonable to reduce the type parameters of `Execution` down to only `C: Parameters`? This means that the `Execution` struct will now only support one type of the proof. Currently in our system, it seems that `S: SNARK` will always be `C::ProgramSNARK`.

Reply:
I agree with this, we should switch `S: SNARK` for `C: Parameters` instead of incurring both. I am happy to make the change and then resample parameters.

Reply:
Right. Should just be `C: Parameters`